drivers/android/context.rs

```diff
@@ -27,20 +31,15 @@ unsafe impl Sync for Context {}
 
 impl Context {
     pub(crate) fn new() -> Result<Arc<Self>> {
-        let mut ctx = Pin::from(UniqueArc::try_new(Self {
-            // SAFETY: Init is called below.
-            manager: unsafe {
-                Mutex::new(Manager {
+        let ctx = UniqueArc::pin_init::<core::convert::Infallible>(pin_init!(Self {
```

Comment:
Currently I have not implemented `InPlaceInit` for `Arc`, since it does not fit the API (you can only call `pin_init` on it). But since we have full control over it, we could just add it as a method. Will do so 👍

Reply:
This is just an unfortunate result of using `?` (which is why I chose to not have `?` in my `pin-init` crate but instead make error propagation automatic).

Basically since all error return paths use `?` which expands to a `From::from`, the error type never unifies with anything and stay as an inference variable, hence the explicit specification.

Reply:
If we had the never type this would be a much more bearable `::<!>`

Reply:
It works with the never type